wordel.el

```diff
@@ -374,16 +374,19 @@ If PROPS are non-nil, they are used in place of default values."
 
 (defun wordel--comparison (guess subject)
   "Return propertized GUESS character list compared against SUBJECT."
-  (cl-loop with guesses =  (split-string guess "" 'omit-nulls)
-           with matches = nil
-           for i from 0
-           for g in guesses
-           for match = (setf (alist-get g matches nil t #'equal)
-                             (string-match-p g subject (1+ (alist-get g matches -1 nil #'equal))))
-           collect (propertize g 'hint (if (not match) 'wordel-guessed
-                                         (if (eq (string-to-char g) (aref subject i))
-                                             'wordel-correct
-                                           'wordel-almost)))))
+  (cl-loop with non-matches = (cl-loop for s across subject
+                                       for g across guess
+                                       unless (= g s)
+                                       collect s)
+           for s across subject
+           for g across guess
+           collect (propertize (string g) 'hint (if (= g s)
```

Comment:
We can use `char-to-string` here since we know we're dealing with individual characters

Reply:
Good idea, fixed now. I felt the line was getting long, though, therefore I introduced two more newlines.